geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy

```diff
@@ -1780,11 +1785,38 @@ String rasterizeIndicators(JdbcDataSource datasource,
         }
         def freeFacadeDensityExact = Geoindicators.RsuIndicators.freeExternalFacadeDensityExact(datasource, createScalesRelationsGridBl,
                 grid, grid_column_identifier, prefixName)
+        tablesToJoinForWidth.put(freeFacadeDensityExact, grid_column_identifier)
         if (freeFacadeDensityExact) {
-            if (list_indicators_upper.intersect(["FREE_EXTERNAL_FACADE_DENSITY", "ASPECT_RATIO"])) {
+            if (list_indicators_upper.intersect(["FREE_EXTERNAL_FACADE_DENSITY", "ASPECT_RATIO", "STREET_WIDTH"])) {
                 indicatorTablesToJoin.put(freeFacadeDensityExact, grid_column_identifier)
                 tablesToDrop << freeFacadeDensityExact
             }
+
+            def gridForWidth = Geoindicators.DataUtils.joinTables(datasource, tablesToJoinForWidth, "grid_for_width")
+            tablesToDrop << gridForWidth
+
+            // Compute the aspect_ratio (and street width if needed)
+            if (list_indicators_upper.intersect(["ASPECT_RATIO", "STREET_WIDTH"]) && building) {
+                datasource "ALTER TABLE $gridForWidth RENAME COLUMN $grid_column_identifier TO ID_RSU"
+                def aspectRatio = Geoindicators.RsuIndicators.aspectRatio(datasource, gridForWidth,
+                        "free_external_facade_density", "building_fraction", prefixName)
+                datasource "ALTER TABLE $aspectRatio RENAME COLUMN ID_RSU TO $grid_column_identifier"
+                indicatorTablesToJoin.put(aspectRatio, grid_column_identifier)
+                tablesToDrop << aspectRatio
+
+                // Calculates the street width if needed
+                if (list_indicators_upper.contains("STREET_WIDTH") && building) {
+                    tablesToJoinForWidth.put(aspectRatio, grid_column_identifier)
+                    gridForWidth = Geoindicators.DataUtils.joinTables(datasource, tablesToJoinForWidth, "grid_for_width")
```

Comment:
if(gridForWidth) then ?

Reply:
No because this joined table contains indicators useful for aspect ratio AND street width